contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,587 @@
+    function withdraw(uint256 _pid, uint256 _amount) external nonReentrant {
+        harvestFromMasterChef();
+        updatePool(_pid);
+        PoolInfo storage pool = poolInfo[_pid];
+        UserInfo storage user = userInfo[_pid][msg.sender];
+
+        if (user.amount > 0) {
+            // Harvest JOE
+            uint256 boostedLiquidity = getBoostedLiquidity(_pid, msg.sender);
+
+            uint256 pending = boostedLiquidity
+                .mul(pool.accJoePerShare)
+                .div(ACC_TOKEN_PRECISION)
+                .sub(user.rewardDebt)
+                .add(claimableJoe[_pid][msg.sender]);
+            claimableJoe[_pid][msg.sender] = 0;
+            JOE.safeTransfer(msg.sender, pending);
+            emit Harvest(msg.sender, _pid, pending);
+        }
+
+        // Effects
+        uint256 boostedLiquidityBefore = getBoostedLiquidity(_pid, msg.sender);
+        user.amount = user.amount.sub(_amount);
+        uint256 boostedLiquidityAfter = getBoostedLiquidity(_pid, msg.sender);
+        // Update the `pool.totalBoostedAmount` and `totalVeJoe` if needed.
+        if (user.amount == 0) {
+            pool.totalVeJoe = pool.totalVeJoe.sub(user.veJoeBalance);
+        }
+
+        pool.totalBoostedAmount = pool.totalBoostedAmount.add(boostedLiquidityAfter).sub(boostedLiquidityBefore);
+        user.rewardDebt = boostedLiquidityAfter.mul(pool.accJoePerShare).div(ACC_TOKEN_PRECISION);
+
+        // Interactions
+        IRewarder _rewarder = pool.rewarder;
+        if (address(_rewarder) != address(0)) {
+            _rewarder.onJoeReward(msg.sender, user.amount);
+        }
+
+        pool.lpToken.safeTransfer(msg.sender, _amount);
+
+        emit Withdraw(msg.sender, _pid, _amount);
+    }
+
+    /// @notice Updates a pool and user boost factors after
+    /// a veJoe token operation. This function needs to be called
+    /// by the veJoe contract after every event.
+    /// @param _user The users address we are updating
+    /// @param _newUserBalance The new balance of the users veJoe
+    function updateBoost(address _user, uint256 _newUserBalance) external {
+        require(msg.sender == address(VEJOE), "BoostedMasterChefJoe: Caller not veJOE");
+
+        uint256 length = poolInfo.length;
+
+        for (uint256 pid = 0; pid < length; ++pid) {
+            UserInfo storage user = userInfo[pid][_user];
+
+            // Skip if user doesn't have any deposit in the pool
+            if (user.amount == 0) {
+                continue;
+            }
+
+            PoolInfo storage pool = poolInfo[pid];
+            updatePool(pid);
+            // calculate pending
+            uint256 oldBoostedLiquidity = getBoostedLiquidity(pid, _user);
+            uint256 pending = oldBoostedLiquidity.mul(pool.accJoePerShare).div(ACC_TOKEN_PRECISION).sub(
+                user.rewardDebt
+            );
+
+            // Increase claimableJoe
+            claimableJoe[pid][_user] += pending;
```

Comment:
Since we're not using 0.8 or higher, change to `claimableJoe[pid][_user] = claimable[pid][_user].add(pending)`

Reply:
Fixed this (and also a related bug in `updateBoost`) and added a test for claimable.